src/main/java/com/telefonica/iot/perseo/TimeRulesStore.java

```diff
@@ -0,0 +1,159 @@
+
+    public static TimeRulesStore getInstance() {
+        if(INSTANCE == null) {
+            INSTANCE = new TimeRulesStore();
+        }
+
+        return INSTANCE;
+    }
+
+    /**
+     * Save 'timed rules' name, service and subservice for later use of this information when
+     * the timed events are launched from GenericListener
+     *
+     * @param body POST or PUT body with rules data
+     */
+    public void saveTimeRules(String body) {
+
+        if (body.equals("")) {
+            return;
+        } else {
+
+            JSONArray jo;
+            try {
+                // For updateAll requests from perseo-fe
+                jo = new JSONArray(body);
+            } catch (JSONException e) {
+                // For new Rules requests from perseo-fe
+                jo = new JSONArray();
+                jo.put(new JSONObject(body));
+            }
+
+            String strName;
+            String ruleText;
+            for (int i = 0; i < jo.length(); i++) {
+
+                try {
+                    strName = (String) jo.getJSONObject(i).get("name");
+                    ruleText = (String) jo.getJSONObject(i).get("text");
+                } catch (JSONException e) {
+                    // Invalid Rule
+                    continue;
+                }
+
+                // Only "Timed Rules"
+                if (!strName.startsWith("ctxt$") && isTimeRule(ruleText)) {
+
+                    // Current active rule. Save header information by name
+                    List<String> ruleName = new ArrayList(Arrays.asList(strName.split("@")));
+                    List<String> context = new ArrayList(Arrays.asList(ruleName.get(1).split("/")));
+                    jo.getJSONObject(i).put("service", context.get(0));
+                    context.remove(0);
+                    jo.getJSONObject(i).put("subservice", "/" + String.join("/", context));
+                    rulesInfo.put(ruleName.get(0), jo.getJSONObject(i));
+                }
+            }
+        }
+    }
+
+    /**
+     * Check if a ruleText is a 'timed rule'
+     *
+     * @param ruleText The rule text
+     * @return true if ruleText is a 'timed rule', false otherwise
+     */
+    private boolean isTimeRule(String ruleText) {
+        // Detect timed rules, searching "timer:XX" patterns or Match_Recognize interval patterns
+        // http://esper.espertech.com/release-6.0.1/esper-reference/html/match-recognize.html#match-recognize-interval
+        // http://esper.espertech.com/release-6.0.1/esper-reference/html/event_patterns.html#pattern-timer-interval
+        // http://esper.espertech.com/release-6.0.1/esper-reference/html/event_patterns.html#pattern-timer-at
+        // http://esper.espertech.com/release-6.0.1/esper-reference/html/event_patterns.html#pattern-timer-schedule
+        return ruleText.toLowerCase().contains(" timer:") ||
+                (ruleText.toLowerCase().contains(" match_recognize") && ruleText.toLowerCase().contains("interval"));
```

Comment:
Is the withespace just before `match_recoginezze` correct in contains() argument? (just checking)

Reply:
Ok.
Better we removed it. Right now it works fine, but if the rules will have line breaks one day, it could become a problem.

Reply:
Fixed in 589775e